backtrace-library/src/main/java/backtraceio/library/common/BatteryStateHelper.java

```diff
@@ -0,0 +1,244 @@
+package backtraceio.library.common;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.os.BatteryManager;
+
+
+import androidx.lifecycle.Lifecycle;
+import androidx.lifecycle.LifecycleObserver;
+import androidx.lifecycle.OnLifecycleEvent;
+import androidx.lifecycle.ProcessLifecycleOwner;
+
+import java.util.HashMap;
+
+import backtraceio.library.BacktraceDatabase;
+import backtraceio.library.interfaces.Database;
+import backtraceio.library.logger.BacktraceLogger;
+
+public class BatteryStateHelper extends BroadcastReceiver implements LifecycleObserver {
+    private static Context context;
+    private static BacktraceDatabase database;
+    private static boolean enabled;
+    private static String batteryHealth, batteryLevel, batteryChargingSource, batteryChargingStatus, batteryTechnology, batteryTemperature;
+    private static IntentFilter intentFilter;
+    private static BatteryStateHelper batteryStateHelper;
+    private static String TAG = "BatteryStateHelper";
+
+    private BatteryStateHelper(Context context, Database database) {
+        if (BatteryStateHelper.context == null){
+            BatteryStateHelper.context = context;
+            if (database != null) {
+                BatteryStateHelper.database = (BacktraceDatabase)database;
+            }
+        }
+        if (intentFilter == null) {
+            intentFilter = new IntentFilter();
+            intentFilter.addAction(Intent.ACTION_BATTERY_CHANGED);
+        }
+    }
+
+    public static BatteryStateHelper getInstance(Context context, Database database) {

```

Comment:
Nit: This doesn't need to be a singleton. It would make your `enable` and `disable` functions cleaner, avoid having to make your functions `static`, and you can just use the keyword `this` when you want to get the current class instance. There should be a good reason for making a class a Singleton.